### ISSUE TITLE
Support for custom types

### ISSUE THREAD

**Issue by solnic** [solnic/drops#9]

**Comment by ilyabayel** [comment]
Hey, @solnic. How're you doing? Does this library support union types? https://en.wikipedia.org/wiki/Tagged_union

Maybe something like:

```elixir
defmodule ContainerContract do
    use Drops.Contract

    schema do
      %{
        required(:container_id) => string(:filled?),
        required(:cargos) =>
          list(
            union([
              %{
                required(:__type) => "Car",
                required(:make) => string(:filled?),
                required(:model) => string(:filled?)
              },
              %{
                required(:__type) => "Animal",
                required(:name) => string(:filled?),
                required(:species) => string(:filled?)
              }
            ])
          )
      }
    end
```

what do you think?

**Comment by solnic** [comment]
This was implemented in #36
